Correct not working examples in differences from R section

bkamins opened the pull request:
1. `[1:4]+[1:2]` in Julia is not equivalent of `c(1,2,3,4)+c(1,2)` in R.
2. I am not aware of `apply` function in base Julia. I would say that the closest equivalent of `lapply` is `map`.

bkamins committed:
Correct not working examples in differences from R

1. `[1:4]+[1:2]` in Julia is not equivalent of `c(1,2,3,4)+c(1,2)` in R.
2. I am not aware of `apply` function in Julia. I would say that the closest equivalent of `lapply` is `map`.

ararslan commented:
The Julia equivalent of `apply` would probably be `mapslices`.

johnmyleswhite commented on doc/manual/noteworthy-differences.rst:
```diff
@@ -140,9 +140,9 @@ noteworthy differences:
-  equivalent ``[1:4] + [1:2]`` will throw an error in Julia.
-- Julia's :func:`apply` takes the function first, then its arguments, unlike
-  ``lapply(<structure>, function, arg2, ...)`` in R.
+  equivalent ``[1,2,3,4] + [1,2]`` will throw an error in Julia.
```
Can you insert spaces after commas to match the style in the rest of the file? Otherwise this is good by me.

bkamins committed:
Added spaces in R differences documentation

I have also added spaces in several places above.

johnmyleswhite reviewed (approved):
Good by me once tests pass.

bkamins commented:
@ararslan Yes, I would say that there is approximately the following mapping:
1. `map` is `lapply`;
2. `mapslices` is `apply`;
3. `broadcast` is `mapply` (interestingly here in R function is a first argument :)).

simonbyrne commented:
@bkamins would you mind adding mentions of those other functions as well?

bkamins committed:
Added mapslices and broadcast description

Added mapslices, broadcast and dot syntax for vectorizing functions description in differences from R section.

simonbyrne commented:
Thanks!

tkelman commented on doc/manual/noteworthy-differences.rst:
```diff
@@ -133,16 +133,21 @@ noteworthy differences:
+  equivalent ``[1, 2, 3, 4] + [1, 2]`` will throw an error in Julia.
+- Julia's :func:`map` takes the function first, then its arguments, unlike
+  ``lapply(<structure>, function, ...)`` in R. Similarly Julia's equivalent of
+  ``apply(X, MARGIN, FUN, ...)`` in R is :func:`mapslices` where function is a
```
where the function is the first argument
